plugins/outputs/azure_monitor_logs/azure_monitor_logs_test.go

```diff
@@ -0,0 +1,277 @@
+package azure_monitor_logs
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/metric"
+	"github.com/stretchr/testify/require"
+)
+
+func getMetric(name string) telegraf.Metric {
+	m, err := metric.New(
+		name,
+		map[string]string{
+			"host": "test-machine",
+			"env":  "testing",
+		},
+		map[string]interface{}{
+			"value": 42.0,
+		},
+		time.Unix(0, 0),
+	)
+	if err != nil {
+		panic(err)
+	}
+	return m
+}
+
+func TestMissingCustomerId(t *testing.T) {
+	plugin := &AzLogAnalytics{}
+
+	err := plugin.Connect()
+	require.Error(t, err)
+}
+
+func TestMissingSharedKey(t *testing.T) {
+	plugin := &AzLogAnalytics{
+		CustomerID: "dummy_id",
+	}
+
+	err := plugin.Connect()
+	require.Error(t, err)
+}
+
+func TestInvalidNamespacePrefix(t *testing.T) {
+	plugin := &AzLogAnalytics{
+		CustomerID:      "dummy_id",
+		SharedKey:       "ZHVtbXlfa2V5",
+		NamespacePrefix: "Testing_",
+	}
+
+	err := plugin.Connect()
+	require.Error(t, err)
+}
+
+func TestInvalidNamespacePrefixLength(t *testing.T) {
+	plugin := &AzLogAnalytics{
+		CustomerID:      "dummy_id",
+		SharedKey:       "ZHVtbXlfa2V5",
+		NamespacePrefix: "ThisIsALongStringToTestTheMaxLength",
+	}
+
+	err := plugin.Connect()
+	require.Error(t, err)
+}
+
+func TestValid(t *testing.T) {
+	plugin := &AzLogAnalytics{
+		CustomerID: "dummy_id",
+		SharedKey:  "ZHVtbXlfa2V5",
+	}
+
+	err := plugin.Connect()
+	require.NoError(t, err)
+}
```

Comment:
This will make an actual connection right? If so, please either setup a http-test-server (see e.g. http input plugin) or wrap this test so it is only executed in long tests, I prefer the first option.